### PR TITLE
SDI-71: 📝 More spring docs configuration

### PR DESCRIPTION
**Pull request by petergphillips**

### CODE REVIEW

File: build.gradle.kts [hunk 1]
```diff
@@ -44,11 +44,10 @@ dependencies {
   implementation("net.sf.ehcache:ehcache:2.10.9.2")
   implementation("com.zaxxer:HikariCP:5.0.1")
 
-  implementation("io.springfox:springfox-boot-starter:3.0.0")
-  implementation("org.springdoc:springdoc-openapi-ui:1.6.1")
-  implementation("org.springdoc:springdoc-openapi-kotlin:1.6.1")
-  implementation("org.springdoc:springdoc-openapi-data-rest:1.6.1")
-  implementation("io.swagger.core.v3:swagger-core:2.1.13")
+  implementation("io.swagger:swagger-annotations:1.6.5")
```
Comment (petergphillips) [inline build.gradle.kts:47]: left in the old annotations for spring fox model items for now

File: build.gradle.kts [hunk 2]
```diff
@@ -113,7 +112,6 @@ tasks {
     }
     minHeapSize = "128m"
     maxHeapSize = "2048m"
-    jvmArgs = listOf("--add-opens", "java.base/java.lang=ALL-UNNAMED", "--add-opens", "java.base/java.util=ALL-UNNAMED")
```
Comment (petergphillips) [inline build.gradle.kts:116]: ignored for java 17 anyway

File: Dockerfile [hunk 3]
```diff
@@ -15,7 +15,6 @@ ENV BUILD_NUMBER ${BUILD_NUMBER:-1_0_0}
 
 RUN apt-get update && \
     apt-get -y upgrade && \
-    apt-get install -y curl && \
```
Comment (petergphillips) [inline Dockerfile:18]: we've started to take this out everywhere

File: src/test/java/uk/gov/justice/hmpps/prison/api/resource/OpenApiDocsTest.kt [hunk 4]
```diff
@@ -91,7 +91,7 @@ class OpenApiDocsTest : ResourceTest() {
       .expectStatus().isOk
       .expectBody()
       .jsonPath("$.components.securitySchemes.bearer-jwt")
-      .isEqualTo(mapOf("type" to "http", "scheme" to "bearer", "bearerFormat" to "JWT", "name" to "Authorization", "in" to "header"))
+      .isEqualTo(mapOf("type" to "http", "scheme" to "bearer", "bearerFormat" to "JWT"))
```
Comment (petergphillips) [inline src/test/java/uk/gov/justice/hmpps/prison/api/resource/OpenApiDocsTest.kt:94]: not sure why this has changed - think it is something to do with taking spring fox out.  The other projects have only the 3 map items anyway so brings it into line with them

Comment (petergphillips) [reply src/test/java/uk/gov/justice/hmpps/prison/api/resource/OpenApiDocsTest.kt:94]: actually will be fixed in next PR

Comment (petergphillips) [reply src/test/java/uk/gov/justice/hmpps/prison/api/resource/OpenApiDocsTest.kt:94]: seems to actually be due to spring docs upgrade from 1.6.1 to 1.6.6

**Review by steverendell — APPROVED** [review]
lgtm

**Pull request merged**